### ISSUE TITLE
Standardize how strings get written to HDF5

### ISSUE THREAD

**Issue by ajtritt** [NeurodataWithoutBorders/pynwb#54]
Strings are written to HDF5 by setting the type as follows:
```python
dt = special_dtype(vlen=str)
```
This results in h5py writing strings to disk as ASCII when running Python 2 and as UTF-8 when running Python3. 

We need to determine a consistent encoding, and update h5ools to ensure the decided-upon encoding always gets used.

**Comment by t-b** [comment]
h5py has some problems with writing string arrays encoded in utf-8. See https://github.com/h5py/h5py/issues/624.

**Comment by oruebel** [comment]
Writing either ASCII or Unicode from both Python 2 and 3 works with h5py. The bug you are refering to  seems to be specific to the case when you use the create_dataset method without specifying a data type, however, the following variants for writing strings work fine:

```
>>> dset.attrs["name_utf8"] = u"Hello"
```

or:

```
>>> dt = h5py.special_dtype(vlen=unicode)
>>> dset = f.create_dataset("name", (100,), dtype=dt)
```

The issue is that the following has different behaviors in Python 2/3:

```
>>> dt = h5py.special_dtype(vlen=str)      # In Python 2 this is ASCII in Python 3 this is UTF8
>>> dset = f.create_dataset("name", (100,), dtype=dt)
```

Similarly

```
dset.attrs["name] = "Hello"    # In Python 2 this is ASCII and in Python 3 this is UTF8
```
To ensure consistent behavior across Python 2/3 and enable users outside of the US to use the format, we should use UTF8 encoding on write in Python 2.

**Comment by oruebel** [comment]
See #13 for some further discussion on string data in HDF5. I closed #13 it seems to largely be a duplicate of this ticket.
